libfreerdp/codec/xcrush.c

```diff
@@ -725,7 +725,7 @@ int xcrush_decompress_l1(XCRUSH_CONTEXT* xcrush, BYTE* pSrcData, UINT32 SrcSize,
 		if ((pSrcData + 2) > pSrcEnd)
 			return -1003;
 
-		MatchCount = *((UINT16*) pSrcData);
+		Data_Read_UINT16(pSrcData, MatchCount);
 
 		MatchDetails = (RDP61_MATCH_DETAILS*) &pSrcData[2];
```

Comment:
Which initialization do you have in mind? `MatchDetails` is used for initialization of `Literals` on the next line and `Literals` is used in the following code, so I don't think we can remove it.